components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/util/SAMLLogoutUtil.java

```diff
@@ -0,0 +1,305 @@
+        return generator.generateIdentifier();
+    }
+
+    /**
+     * Extract the required federated identity provider's configurations into a map.
+     *
+     * @param identityProvider {@link IdentityProvider}Federated identity provider.
+     * @return Map<String, String> Map of properties of the Identity Provider.
+     */
+    public static Map<String, String> getFederatedIdPConfigs(IdentityProvider identityProvider) {
+
+        Property[] properties = identityProvider.getDefaultAuthenticatorConfig().getProperties();
+        List<String> idpPropertyNames = Arrays.asList(SP_ENTITY_ID, SSO_URL, IS_AUTHN_RESP_SIGNED,
+                INCLUDE_CERT, IS_LOGOUT_REQ_SIGNED, IS_SLO_REQUEST_ACCEPTED);
+
+        return Arrays.stream(properties)
+                .filter(t -> idpPropertyNames.contains(t.getName()))
+                .collect(Collectors.toMap(Property::getName, Property::getValue));
+    }
+
+    /**
+     * Build error response when request contain validation or processing errors.
+     *
+     * @param samlMessageContext {@link SAMLMessageContext} object which holds details on logout flow.
+     * @param inResponseTo       ID of the Logout Request.
+     * @param statusCode         Status Code of the Error Response.
+     * @param statusMsg          Status Message of the Error Response.
+     * @return String            Encoded Error Response.
+     * @throws SAMLLogoutException Error when building the Error Response.
+     */
+    public static String buildErrorResponse(SAMLMessageContext samlMessageContext, String
+            inResponseTo, String statusCode, String statusMsg) throws SAMLLogoutException {
+
+        try {
+            LogoutResponse errorResponse = buildResponse(samlMessageContext, inResponseTo, statusCode, statusMsg);
+            return SSOUtils.encode(SSOUtils.marshall(errorResponse));
+        } catch (SAMLSSOException e) {
+            throw new SAMLLogoutException("Error Serializing the SAML Response", e);
+        }
+    }
+
+    /**
+     * Build the Logout Response for logout request from the federated IdP.
+     *
+     * @param samlMessageContext {@link SAMLMessageContext} object which holds details on logout flow.
+     * @param inResponseTo       ID of the Logout Request.
+     * @param statusCode         Status Code of the Error Response.
+     * @param statusMsg          Status Message of the Error Response.
+     * @return Logout Response   Built Logout Response.
+     * @throws SAMLLogoutException Error when building the Logout Response.
+     */
+    public static LogoutResponse buildResponse(SAMLMessageContext samlMessageContext, String inResponseTo,
+                                               String statusCode, String statusMsg) throws SAMLLogoutException {
+
+        try {
+            doBootstrap();
+            String issuerID = (String) samlMessageContext.getFedIdPConfigs().get(SP_ENTITY_ID);
+            String acsUrl = (String) samlMessageContext.getFedIdPConfigs().get(SSO_URL);
```

Comment:
Don't we need a null check for `samlMessageContext.getFedIdPConfigs()`?

Reply:
Fixed with PR #133